### PR TITLE
Upgraded .NET Core version to 3.1

### PR DESCRIPTION
**Pull request by Blackbaud-ValKuppuswamy**

### CODE REVIEW

File: sky-api-tutorial-auth-code-c-sharp.csproj [hunk 1]
```diff
@@ -21,9 +20,10 @@
     <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.0.4" />
```
Comment (Blackbaud-ChristiSchneider) [inline sky-api-tutorial-auth-code-c-sharp.csproj:20]: Should these all be updated? This package has version 2.2.0 out now.

Comment (Blackbaud-ValKuppuswamy) [reply sky-api-tutorial-auth-code-c-sharp.csproj:20]: I was wondering about that too. Let me upgrade them as well.

Comment (Blackbaud-ChristiSchneider) [reply sky-api-tutorial-auth-code-c-sharp.csproj:20]: I would just go ahead and update everything to the latest that is supported by .NET 3.1. You can check on nuget.org if you expand dependencies on the latest version. Only thing to watch for is that you don't want a version that only supports 5.0, so you'd have to fall back to the previous major version of any library like that.

**Comment by Blackbaud-ChristopherRodgers** [comment]
Can you perform the changes to Program.cs described in https://docs.microsoft.com/en-us/aspnet/core/migration/1x-to-2x/?view=aspnetcore-5.0#update-main-method-in-programcs ? 

It seems the move to 3.0 revised Program.cs a little more...
https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-5.0&tabs=visual-studio#hostbuilder-replaces-webhostbuilder

At some point, I wonder if it just makes sense to start a new project from the .NET Core 5.0 template (so we're on the latest released version) and copy over the controller endpoints, services, views, and app settings (if that's easier than in-place upgrading this project).

If that's too extreme, we can worry about doing a 5.0 from template later.

**Comment by Blackbaud-ChristiSchneider** [comment]
We use .NET 3.1 internally since it's in LTS and 5.0 is not (5.1 will be, in probably a little less than a year). I _think_ it makes sense to do the same for examples since others may not want to use a version that isn't in LTS. I agree it would probably be best to start from the 3.1 template and just copy in the logic to a new project.

**Comment by Blackbaud-ChristopherRodgers** [comment]
Ahh, good call. Didn't realize [End of Support for 5.0 ](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) was earlier than 3.1.

File: sky-api-tutorial-auth-code-c-sharp.csproj (hunk 1 above)
Comment (Blackbaud-ValKuppuswamy) [reply sky-api-tutorial-auth-code-c-sharp.csproj]: Updated the packages

File: sky-api-tutorial-auth-code-c-sharp.csproj [hunk 2]
```diff
@@ -17,24 +16,24 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Server.IISIntegration" Version="2.2.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="5.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel.Https" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Http" Version="2.2.2" />
+    <PackageReference Include="Microsoft.AspNetCore.Session" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="5.0.0" />
```
Comment (Blackbaud-ChristopherRodgers) [inline sky-api-tutorial-auth-code-c-sharp.csproj:29]: Are these `Microsoft.Extensions.Configuration` packages appropriate for .Net Core 3.1? I see that [3.13](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Json/3.1.13) is more recent. The [5.0 version](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Json/5.0.0) references .Net 5.0 as a dependency.

Comment (Blackbaud-ValKuppuswamy) [reply sky-api-tutorial-auth-code-c-sharp.csproj]: Good catch. Downgraded the packages to supported version

**Review by Blackbaud-ChristopherRodgers — APPROVED** [review]

**Pull request merged**